src/gridcoin/voting/result.cpp

```diff
@@ -633,7 +633,21 @@ class LegacyVoteCounterContext
     const LegacyChoiceMap& GetChoices()
     {
         if (m_legacy_choices_cache.empty()) {
-            for (uint8_t i = 0; i < m_poll.Choices().size(); ++i) {
+            uint8_t m_poll_choices_size = 0;
+
+            // This silences the code scan warning of a possible infinite loop due to the overflow of the uint8_t i counter
+            // against the comparison to m_poll.Choices.size(). Note it is the logged equivalent to std::min<uint8_t>. Since
+            // this is legacy polls only and those are no longer issued, and this condition does not actually exist in the
+            // chain, this is purely for formality's sake.
+            if (m_poll.Choices().size() > (size_t) std::numeric_limits<uint8_t>::max()) {
+                LogPrintf("WARN: %s: Number of legacy poll choices exceeds bins available in m_legacy_choices_cache map. "
+                          "Limiting to %u.", __func__, std::numeric_limits<uint8_t>::max());
+                m_poll_choices_size = (size_t) std::numeric_limits<uint8_t>::max();
```

Comment:
nit:
```suggestion
                m_poll_choices_size = std::numeric_limits<uint8_t>::max();
```
Maybe even just `m_poll_choices_size = std::min(m_poll.Choices().size(), std::numeric_limits<uint8_t>::max());`.

Reply:
Yeah. The implicit cast is good enough...

Reply:
I am a big fan of explicit casting, even where it is not needed. Implicit casting has gotten many developers in trouble...

Reply:
I also like explicit casting, just casting it to `size_t` there doesn't make sense when you are assigning to a `uint8_t` or when the type of `std::numeric_limits<uint8_t>::max()` is already uint8_t.

Reply:
Ah true.